tests/testutils/httputils.go

```diff
@@ -44,6 +43,8 @@ func HttpServe(addr string, timeout int) error {
 
 	c := make(chan string)
 	go func() {
+		log.Printf("%v: serving on %v\n", hostname, addr)
```

Comment:
I started with the mentioned interface and a type that wraps `log`. It's not perfect but it's a step forward.